apps/submission/tests/io/test_archive.py

```diff
@@ -187,3 +199,55 @@ def test_parse_meta(self):
             ]
         }
         self.assertDictEqual(archive.meta, expected)
+
+    def test_save(self):
+
+        archive = PixelArchive(self.valid_archive_path)
+        archive.parse()
+
+        self.assertEqual(Pixel.objects.count(), 0)
+        self._load_cgd_entries()
+        archive.save(pixeler=self.pixeler)
+        self.assertEqual(Pixel.objects.count(), 3716)
+        self.assertEqual(Experiment.objects.count(), 1)
+        self.assertEqual(Analysis.objects.count(), 1)
+
+    def test_save_with_existing_experiment_and_analysis(self):
+
+        archive = PixelArchive(self.valid_archive_path)
+        archive.parse()
+        self._load_cgd_entries()
+
+        # Create experiment & analysis
+        experiment = ExperimentFactory(
+            description=archive.meta['experiment']['summary'],
+            omics_area=archive.meta['experiment']['omics_area'],
+            completed_at=archive.meta['experiment']['completion_date'],
+            released_at=archive.meta['experiment']['release_date'],
+        )
+        entry = EntryFactory(
+            identifier=archive.meta['experiment']['entry'],
+            repository=archive.meta['experiment']['repository']
+        )
+        experiment.entries.add(entry)
+        analysis = AnalysisFactory(
+            description=archive.meta['analysis']['description'],
+            secondary_data__from_path=archive.meta['analysis']['secondary_data_path'],  # noqa
+            notebook__from_path=archive.meta['analysis']['notebook_path'],
+            pixeler=self.pixeler,
+            completed_at=archive.meta['analysis']['date'],
+        )
+        analysis.experiments.add(experiment)
+
+        self.assertEqual(Experiment.objects.count(), 1)
+        self.assertEqual(Analysis.objects.count(), 1)
+
+        # Import pixels
+        archive.save(pixeler=self.pixeler)
+        self.assertEqual(Pixel.objects.count(), 3716)
+        for analysis in Analysis.objects.all():
+            print(analysis.__dict__)
```

Comment:
`print`? 🤔

Reply:
Good catch! 😅 I've removed this ugly statement in #111